### PR TITLE
chore: upgrade tooling

### PR DESCRIPTION
**Pull request by ro-ka**
### Description

Upgrade all the tooling! No more audit errors!

bildschirmfoto 2018-08-01 um 16 00 27

### Checklist

- [x] All tests passing
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary
- [x] Commits and PR follow conventions

### CODE REVIEW

File: example/src/app.js [hunk 1]
```diff
@@ -55,7 +55,8 @@ class App extends React.Component {
       {label: 'Tokyo', location: {lat: 35.673343, lng: 139.710388}}
     ];
 
-    return ( // eslint-disable-line
+    return (
+      // eslint-disable-line
```
Comment (yfr) [inline example/src/app.js]: Why disable an empty line?

File: package.json [hunk 2]
```diff
@@ -63,17 +63,18 @@
...
     "js:browser": "browserify src/Geosuggest.jsx --standalone Geosuggest --exclude react -t babelify --extension=.jsx -t browserify-global-shim > dist/react-geosuggest.js",
     "js:browser:uglify": "uglifyjs dist/react-geosuggest.js -o dist/react-geosuggest.min.js -c warnings=false,drop_console=true --mangle",
+    "js:prettier:ci": "prettier --config ./.prettierrc.yml --ignore-path ./.prettierignore --list-different \"**/**.{js,jsx}\"",
+    "js:prettier:fix": "prettier --config ./.prettierrc.yml --ignore-path ./.prettierignore --write \"**/**.{js,jsx}\"",
     "build:module": "babel src --out-dir module && copyfiles -f src/geosuggest.css module/",
     "build:browser": "npm run js:browser && npm run js:browser:uglify",
     "build:example": "npm run prepare:example && npm run js:example:uglify",
+    "release:patch": "npm version patch -m \"chore(release): version %s\"",
+    "release:minor": "npm version minor -m \"chore(release): version %s\"",
+    "release:major": "npm version major -m \"chore(release): version %s\"",
```
Comment (yfr) [inline package.json]: Did not know you can add a commit message to npm version call.

File: src/Geosuggest.jsx [hunk 3]
```diff
@@ -36,10 +36,22 @@ class Geosuggest extends React.Component {
 
     this.onInputChange = this.onInputChange.bind(this);
     this.onAfterInputChange = this.onAfterInputChange.bind(this);
+    this.onInputFocus = this.onInputFocus.bind(this);
```
Comment (yfr) [inline src/Geosuggest.jsx]: Why move all class properties with arrow functions to calls methods?

Functions as properties have `this` bound automatically. So no need to bind them explicitly.

File: src/Geosuggest.jsx [hunk 4]
```diff
@@ -201,20 +226,20 @@ class Geosuggest extends React.Component {
...
-                this.activateSuggest('next');
-              }
-            });
-        }
-      );
+      this.autocompleteService.getPlacePredictions(options, suggestsGoogle => {
+        this.setState({isLoading: false});
+        this.updateSuggests(
+          suggestsGoogle || [], // can be null
+          () => {
```
Comment (yfr) [inline src/Geosuggest.jsx]: Move this function to a method and avoid creating a new function on every `searchSuggest` call.

File: src/Geosuggest.jsx [hunk 5]
```diff
@@ -414,58 +438,66 @@ class Geosuggest extends React.Component {
    */
   render() {
     const attributes = filterInputAttributes(this.props),
-      classes = classnames(
-        'geosuggest',
-        this.props.className,
-        {'geosuggest--loading': this.state.isLoading}
-      ),
+      classes = classnames('geosuggest', this.props.className, {
```
Comment (yfr) [inline src/Geosuggest.jsx]: I prefer to have `const` for every new variable. It's more readable for me.

just my preference :)

File: src/Geosuggest.jsx [hunk 6]
```diff
@@ -414,58 +438,66 @@ class Geosuggest extends React.Component {
...
+          ref={i => (this.input = i)}
+          value={this.state.userInput}
+          doNotSubmitOnEnter={!this.state.isSuggestsHidden}
+          ignoreTab={this.props.ignoreTab}
+          ignoreEnter={this.props.ignoreEnter}
+          style={this.props.style.input}
+          onChange={this.onInputChange}
+          onFocus={this.onInputFocus}
+          onBlur={this.onInputBlur}
+          onKeyDown={this.props.onKeyDown}
```
Comment (yfr) [inline src/Geosuggest.jsx]: How about deconstructing the `this.props` and `this.state` at the beginning of `render`? Makes for better readability in the components.

File: example/src/app.js (hunk 1 above)
Comment (ro-ka) [reply example/src/app.js]: Reformatted from prettier.

File: src/Geosuggest.jsx (hunk 3 above)
Comment (ro-ka) [reply src/Geosuggest.jsx]: *prettier* didn’t work with the arrow functions.

File: src/Geosuggest.jsx (hunk 4 above)
Comment (ro-ka) [reply src/Geosuggest.jsx]: This would be a refactoring, not just a tooling upgrade.

File: src/Geosuggest.jsx (hunk 5 above)
Comment (ro-ka) [reply src/Geosuggest.jsx]: This would be a refactoring, not just a tooling upgrade.

File: src/Geosuggest.jsx (hunk 6 above)
Comment (ro-ka) [reply src/Geosuggest.jsx]: This would be a refactoring, not just a tooling upgrade.

**Pull request merged**